src/ess/amor/load.py

```diff
@@ -22,54 +30,141 @@ def load_detector(
+) -> RawEvents[Run]:
+    detector_numbers = sc.arange(
+        'event_id',
+        start=0,
+        stop=(Detector.nBlades * Detector.nWires * Detector.nStripes).value,
+        unit=None,
+        dtype='int32',
+    )
+    data = (
+        nexus.extract_detector_data(detector)
+        .bins.constituents['data']
+        .group(detector_numbers)
+        .fold(
+            'event_id',
+            sizes={
+                'blade': Detector.nBlades,
+                'wire': Detector.nWires,
+                'stipe': Detector.nStripes,
+            },
+        )
+    )
     # Recent versions of scippnexus no longer add variances for events by default, so
     # we add them here if they are missing.
-    data = nexus.extract_detector_data(detector)
     if data.bins.constituents['data'].data.variances is None:
         data.bins.constituents['data'].data.variances = data.bins.constituents[
             'data'
         ].data.values
+    x, y, z, angle_from_center_of_beam = pixel_coordinate_in_lab_frame(
+        data.coords['event_id'],
+        detector_rotation,
+    )
+    data.coords['position'] = sc.spatial.as_vectors(x, y, z).to(unit='m')
+    data.coords['angle_from_center_of_beam'] = angle_from_center_of_beam
     return RawEvents[Run](data)
 
 
 def compute_tof(
-    events: RawEvents[Run], phase: ChopperPhase[Run], frequency: ChopperFrequency[Run]
+    data: RawEvents[Run], phase: ChopperPhase[Run], frequency: ChopperFrequency[Run]
 ) -> ChopperCorrectedTofEvents[Run]:
-    data = events.copy(deep=False)
-    dim = 'tof'
-    data.bins.coords[dim] = data.bins.coords.pop('event_time_offset').to(
-        unit='us', dtype='float64', copy=False
+    data.bins.coords['tof'] = data.bins.coords.pop('event_time_offset').to(
+        unit='ns', dtype='float64', copy=False
     )
-    tof_unit = data.bins.coords[dim].bins.unit
+
+    tof_unit = data.bins.coords['tof'].bins.unit
     tau = sc.to_unit(1 / (2 * frequency), tof_unit)
     tof_offset = tau * phase / (180.0 * sc.units.deg)
-    # Make 2 bins, one for each pulse
-    edges = sc.concat([-tof_offset, tau - tof_offset, 2 * tau - tof_offset], dim)
-    data = data.bin({dim: sc.to_unit(edges, tof_unit)})
-    # Make one offset for each bin
-    offset = sc.concat([tof_offset, tof_offset - tau], dim)
-    # Apply the offset on both bins
-    data.bins.coords[dim] += offset
-    # Rebin to exclude second (empty) pulse range
-    data = data.bin({dim: sc.concat([0.0 * sc.units.us, tau], dim)})
+
+    event_time_offset = data.bins.coords['tof']
+
+    minimum = -tof_offset
+    frame_bound = tau - tof_offset
+    maximum = 2 * tau - tof_offset
+
+    offset = sc.where(
+        (minimum < event_time_offset) & (event_time_offset < frame_bound),
+        tof_offset,
+        sc.where(
+            (frame_bound < event_time_offset) & (event_time_offset < maximum),
+            tof_offset - tau,
+            0.0 * tof_unit,
```

Comment:
It's basically the 'outside of frame' case, in that region the `tof` is not defined, events in that region is masked in the line below.

Maybe it would be better to use `nan`, but it's nice to not use `nan` because then you see the events in the figures, but masked.